src/diagapp.jl

```diff
@@ -1,25 +1,37 @@
 # Diagonal Approximation for trace of inverse of matrix
+# Values from Extrapolation Methods for Estimating Trace of the Matrix Inverse by P. Fika
 
-export diagapp
+export diagapp, diagapp!
 
 using LinearAlgebra
 
+struct diagappspace
+    A::AbstractArray{<:Any, 2}
+    x::AbstractArray{<:Any, 1}
+    y::AbstractArray{<:Any, 1}
+end
+
+function diagappspace(A::AbstractArray{<:Any, 2})
+    x = rand(-1.0:2.0:1.0, size(A)[1])
+    y = similar(x, Float64)
```

Comment:
No need for `Float64` here, assuming `A` has some floating point `eltype`. If `eltype(A) <: Integer`, fall back on `Float64`.

Reply:
This is to support `Float32` and higher precision number types.